homeassistant/components/media_player/browse_media.py

```diff
@@ -38,7 +42,21 @@ def async_process_play_media_url(
 
     # convert relative URL to absolute URL
     if media_content_id[0] == "/" and not allow_relative_url:
-        media_content_id = f"{get_url(hass)}{media_content_id}"
+        if (
+            for_addon
+            and hass.config.api
+            and not hass.config.api.use_ssl
+            and hass.components.hassio.is_hassio()
+        ):
+            addon_url = yarl.URL.build(
+                scheme="http",
+                host=hass.components.hassio.get_host_info()["hostname"],
+                port=hass.config.api.port,
+            )
+            media_content_id = f"{str(addon_url)}{media_content_id}"
```

Comment:
No need to cast to string inside string interpolation.
```suggestion
            media_content_id = f"{addon_url}{media_content_id}"
```

Reply:
If I recall correctly, an add-on talking to HA via the supervisor automatically have authentication attached. In that case we don't need to sign the path above.

Reply:
I'm not actually returning the url for supervisor here, I'm returning the hostname of HA. I know that usually addons talk to HA via supervisor but I can't do that in this case. Because for an addon to talk to HA via supervisor it has to include the `SUPERVISOR_TOKEN` we gave it in the header and I can't tell it to do that here, I can only give it a URL. So I have to give it `http://homeassistant:8123` (or whatever the hostname and port is).

Reply:
That's also why this logic is skipped if ssl is turned on since I assume the certificate won't be valid and I can't tell the addon not to verify it.